server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1051,7 +1056,29 @@ private DeleteResult cleanupStaleIndices(Map<String, BlobContainer> foundIndices
             assert false : e;
             logger.warn(new ParameterizedMessage("[{}] Exception during cleanup of stale indices", metadata.name()), e);
         }
-        return deleteResult;
+    }
+
+    private void executeOneStaeIndexDelete(BlockingQueue<Map.Entry<String, BlobContainer>> staleIndicesToDelete, Executor executor,
+                                           GroupedActionListener<DeleteResult> listener) throws InterruptedException {
+        Map.Entry<String, BlobContainer> indexEntry  = staleIndicesToDelete.poll(0L, TimeUnit.MILLISECONDS);
+        if (indexEntry == null) {
+            listener.onResponse(DeleteResult.ZERO);
+        } else {
+            final String indexSnId = indexEntry.getKey();
+            executor.execute(ActionRunnable.supply(listener, () -> {
```

Comment:
I think we can just use `threadPool.executor(ThreadPool.Names.SNAPSHOT)` here instead of passing the `executor` as a parameter down here.

Reply:
Agree. Updated code to use ```threadPool.executor(ThreadPool.Names.SNAPSHOT) ``` instead of executor